app/assets/javascripts/Components/Result/url_viewer.jsx

```diff
@@ -3,19 +3,82 @@ import React from "react";
 export class URLViewer extends React.Component {
   constructor(props) {
     super(props);
-    //TODO: Remove state url and replace its usage with url props when backend is working
     this.state = {
-      url: "https://www.youtube.com/embed/dQw4w9WgXcQ", // Sample Youtube embed link
-      //url: "https://drive.google.com/file/d/1yQh8bzo4IzmtSt-qZ0Vb-BzzEXsp45v_/preview" // sample google drive embed link
-      //url: "https://play.library.utoronto.ca/watch/4493e554354a7e51bdf90f3197185b5f" // sample mymedia link
+      url: "",
+      show_iframe_preview: false,
     };
   }
 
+  componentDidMount() {
+    try {
+      const url = new URL(this.props.content);
+      const youtube_id_is_set = this.configureYoutubePreview(url.toString());
+      if (!youtube_id_is_set) {
+        switch (url.hostname) {
+          case "docs.google.com":
+          case "drive.google.com":
+            this.configureGoogleDrivePreview(url.toString());
+            break;
+          case "play.library.utoronto.ca":
```

Comment:
Yeah, it might be something to do later on. I've been having trouble getting the play.library.utoronto.ca embed player to work properly (namely because I think the code for their embed doesn't resize the video properly) and thus, support to account for it should be added in a later PR.